airbrake/utils.py

```diff
@@ -16,6 +17,16 @@
     TypeType = type
 
 
+class FailProofJSONEncoder(json.JSONEncoder):
+    """Uses object's representation for unsupported types."""
+
+    def default(self, o):  # pylint: disable=E0202
+        # E0202 ignored in favor of compliance with documentation:
+        # https://docs.python.org/2/library/json.html#json.JSONEncoder.default
+        """Return object's repr when not JSON serializable."""
+        return repr(o)
```

Comment:
I noticed that the python docs say

> To extend this to recognize other objects, subclass and implement a default() method with another method that returns a serializable object for o if possible, otherwise it should call the superclass implementation (to raise TypeError).

And their example of a custom `default()` looks like this:

```python
def default(self, o):
   try:
       iterable = iter(o)
   except TypeError:
       pass
   else:
       return list(iterable)
   # Let the base class default method raise the TypeError
   return JSONEncoder.default(self, o)
```

So, I think your custom encoder should change to:

```python
class FailProofJSONEncoder(json.JSONEncoder):
    def default(self, o):
        try:
            return repr(o)
        except Exception:
            return super(FailProofJSONEncoder, self).default(o)
```

In case the call to `repr()` fails.